api/routes/collections.js

```diff
@@ -0,0 +1,207 @@
+    values: [req.params.collection_id]
+  }
+
+  // Run query
+  db.client.query(query, (err, result) => {
+    if (err) {
+      return res.status(400).send(err);
+    } else if (!result.rows.length) {
+      err = { "Error": "No collection with this id exists" };
+      return res.status(404)
+        .set({ "Content-Type": "application/json" })
+        .send(err);
+    }
+
+    let collection = apiUtils.formatCollectionGames(result.rows, hostname);
+      res.status(200)
+        .set({ "Content-Type": "application/json" })
+        .send(collection);
+  });
+
+});
+
+/* Update a collection */
+router.patch('/:collection_id', (req, res) => {
+  let hostname = req.protocol + '://' + req.headers.host;
+
+  let query = { text: '', values: [] };
+  let query_fields = [];
+
+  if ("name" in req.body) {
+    query.values.push(req.body.name);
+    query_fields.push('name = $' + query.values.length);
+  }
+
+  if ("isPrivate" in req.body) {
+    query.values.push(req.body.isPrivate);
+    query_fields.push('"isPrivate" = $' + query.values.length);
+  }
+
+  // Check if no valid fields
+  if (!query_fields.length) {
+    let err = { "Error": "The request object is missing at least one valid field" };
+    return res.status(400)
+      .set({ "Content-Type": "application/json" })
+      .send(err);
+  }
+
+  query.values.push(parseInt(req.params.collection_id));
+  query.text = 'UPDATE "UserCollection" SET ' + query_fields.join(', ') + ' WHERE id = $' + query.values.length + ' RETURNING *';
+
+  // Run query
+  db.client.query(query, (err, result) => {
+    if (err) {
+      return res.status(400).send(err);
+    } else if (!result.rows.length) {
+      err = { "Error": "No collection with this id exists" };
+      return res.status(404)
+        .set({ "Content-Type": "application/json" })
+        .send(err)
+    }
+
+    let collection = apiUtils.formatCollection(result.rows[0], hostname);
+    res.status(200)
+      .set({ 
+        "Content-Type": "application/json",
+        "Content-Location": collection.url
+      })
+      .send(collection);
+  });
+
+});
+
+/* Add a game to a collection */
+router.put('/:collection_id/games/:game_id', (req, res) => {
+
+  // Build query object
+  let query = { 
+    text: 'INSERT INTO "CollectionGame"("userColllectionID", "gameID") SELECT $1, $2 WHERE NOT EXISTS (SELECT "userColllectionID", "gameID" FROM "CollectionGame" WHERE "userColllectionID" = $1 AND "gameID" = $2) RETURNING *', 
+    values: [req.params.collection_id, req.params.game_id] 
+  };
+
+  // Run query
+  db.client.query(query, (err, result) => {
+    if (err) {
+      return res.status(400).send(err);
+    } else if (!result.rows.length) {
+      err = { "Error": "This game is already part of this collection" };
+      return res.status(404)
+        .set({ "Content-Type": "application/json" })
```

Comment:
This is interesting. Why do you set the content-type of the response body? From what I remember, the error should be sent back out in a readable format to the client without doing it

Reply:
This is what we had learned to do in CS493. It's probably more relevant for public-facing APIs, where a certain response format is expected? In any case, I'm open to changing this

Reply:
Interesting. Good to know. I'm fine with it as is and was just curious